src/matchers/to-be-visible.tsx

```diff
@@ -0,0 +1,64 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { matcherHint } from 'jest-matcher-utils';
+import { StyleSheet } from 'react-native';
+import { getHostParent } from '../helpers/component-tree';
+import { checkHostElement, formatElement } from './utils';
+
+function isVisibleForStyles(element: ReactTestInstance) {
+  const style = element.props.style || {};
+  const { display, opacity } = StyleSheet.flatten(style);
+  return display !== 'none' && opacity !== 0;
+}
+
+function isVisibleForAccessibility(element: ReactTestInstance) {
+  return (
+    !element.props.accessibilityElementsHidden &&
+    element.props.importantForAccessibility !== 'no-hide-descendants' &&
+    !element.props['aria-hidden']
```

Comment:
Afaik `aria-hidden` is not passed to host view but translated into `accessibilityElementsHidden` for ios and `importantForAccessibility` so we should be able to skip this in check and still pass the tests.

Reply:
Makes perfect sense - thanks!

Reply:
@mdjastrzebski You're right, but the test still fails when rendering a `<View />`, because RN only translates `aria-hidden` to the accessibility props when rendering the native component itself, which does not get caught by the test renderer

#### View.js
```jsx
      <ViewNativeComponent
        ...
        accessibilityElementsHidden={ariaHidden ?? accessibilityElementsHidden}
        importantForAccessibility={
          ariaHidden === true
            ? 'no-hide-descendants'
            : importantForAccessibility
        }
        ...
      />
```

My second choice would be to use `<Pressable />`, since we're already using it in this test file, but it also fails because of a bug in RN, I believe. This component accepts `aria-hidden` as a prop but only passes it down to the underlying `View`.

In contrast, using any `Touchable` makes the test pass, because their implementation also translates the `aria-hidden` prop into the accessibility props, and they do not render a native component directly, which makes the test renderer aware of these props.

What do you think? Should we use one of the `Touchable`s or you prefer not to rely this much on RN's implementation details?

Reply:
@thiagobrez Can you post `screen.toJSON()` result when rendering `View` with `aria-hidden` prop?

Reply:
I am really surprised now 👀

Reply:
Pls restore `aria-hidden` check then